[GSoC prerequisite] Added 274-Real-ESRGAN notebook

aadhamm opened the pull request:
This tutorial demonstrates step-by-step instructions on converting the Real-ESRGAN Pytorch model to ONNX, then to OpenVINO, running, and comparing the performance of each one.

The tutorial consists of the following steps:

Prepare model settings
Download the Pytorch model (.pth file)
Convert PyTorch model to ONNX
Convert ONNX model to OpenVINO IR
Validate converted model
comparison of Pytorch, ONNX, and OpenVINO IR model's performance

Thank you.

adrianboguszewski commented:
Thank you for your contribution. Before we do the review please do the self-check first (tick the item when it's done):
- [ ] each function is described by docstrings and type hints
- [ ] notebook contains explicit descriptions and explanatory diagrams
- [ ] the notebook doesn't use any data (image, video, etc.) that is not CC4.0 licensed 
- [ ] there is a README.md file in consistent style (look at other notebooks)
- [ ] the notebook is added to the main README and notebooks/README files
- [ ] there are no grammar, punctuation or typo issues (use any free tool for that e.g. Grammarly)
- [ ] there are no committed files besides notebook and readme (please use images or videos from data dir)
- [ ] your PR doesn't change any other notebooks
- [ ] all CI checks passed

aadhamm commented:
- [x]  each function is described by docstrings and type hints
- [x]  notebook contains explicit descriptions and explanatory diagrams
- [x]  the notebook doesn't use any data (image, video, etc.) that is not CC4.0 licensed
- [x]  there is a README.md file in consistent style (look at other notebooks)
- [x]  the notebook is added to the main README and notebooks/README files
- [x]  there are no grammar, punctuation or typo issues (use any free tool for that e.g. Grammarly)
- [x]  there are no committed files besides notebook and readme (please use images or videos from data dir)
- [x]  your PR doesn't change any other notebooks
- [ ]  all CI checks passed

the check keeps saying I have changed other notebooks although I haven't touched them

aadhamm commented:
I will work on the unrequired changes and commit ASAP @adrianboguszewski

aadhamm commented:
Dear @adrianboguszewski, 

**- I have removed the model folder, committing just the notebook and readme**
- [x]  there are no committed files besides notebook and readme (please use images or videos from data dir)

**- resolved the changes in Notebook 102, although, notebook 219 seems to be modified by me although i hadn't touch the "conve_fp32.onnx" file**
- [x]  your PR doesn't change any other notebooks

**also, I wanted to clarify that I modified the "requirements.txt" file to add a necessary dependency, as stated in CONTRIBUTING.md**

Thank you.

adrianboguszewski commented:
@aadhamm, just please revert all changes to conve_fp32.onnx

aadhamm commented:
Dear @adrianboguszewski 

I have tried to revert it in many ways but it didn't work out for this specific file, so I did some workaround and fortunately, it has paid off :) 

I hope it's ready for review.

Thank you so much and sorry for the late reply.

aadhamm commented:
Dear @adrianboguszewski, although I am very satisfied with what I reached in this notebook, I had concerns about maybe it overlaps with existing notebooks that aim for the same target of super-resolution/image restoration and I have the will to add a contribution to OpenVINO. 

I may suggest adding inference for the same model (Real-ESRGAN) but for the scope of Animation Videos or maybe look into other models that scope on old photos restoration if possible.

If you have for me any advice or suggestion, I will appreciate it a lot.

adrianboguszewski commented:
@aadhamm, let's stay with that model but using animation videos would be great! However, we must be sure we have a proper license to use it, so please find a video with a permissive license, ok?

aadhamm commented:
@adrianboguszewski Very well!

Will work on it and update you as soon as i can.

Thank you.

eaidova commented on notebooks/274-Real-ESRGAN-image-restoration/274-Real-ESRGAN-image-restoration.ipynb:
```diff
@@ -0,0 +1,740 @@
+{
```
<p>&gt;Similar to ESRGAN, the modle adopted DIV2K, Flickr2K and OutdoorSceneTraining datasets for training. The training HR patch size is set to 256. Models are trained with four NVIDIA V100 GPUs with a total batch size of 48. Adam optimizer employed.</p><p> </p><p>model -&gt; model?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/939/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

eaidova replied:
<pre class="ql-syntax" spellcheck="false">Line #1.    # Construct the command for Model Optimizer.
</pre><p>could you please use Model Optimizer Python API instead of CLI, please check https://docs.openvino.ai/2023.0/openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html for details</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/939/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

eaidova replied:
<pre class="ql-syntax" spellcheck="false">Line #3.    orig_img = mpimg.imread("../data/image/lr_face.png")
</pre><p>what is the source of input image? Can we use one of provided in this our repo instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/939/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

eaidova replied:
<p>Could you please provide also visualization of openvino results? for me, as user, it is important to see that result is the same like in torch and can be postprocessed using the same way, but I can not find this in your tutorial. Could you please add?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/939/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

eaidova commented on requirements.txt:
```diff
@@ -47,6 +47,7 @@ shapely>=1.7.1
 pyclipper>=1.2.1
 psutil
 gdown
+basicsr
```
could you please install packages related only to your tutorial inside tutorial in prerequisites section using following command:
`
!pip install -q basicsr
`

adrianboguszewski commented:
I'm closing this due to inactivity for a long time. Nevertheless, thank you for contributing and applying for Google Summer of Code. We encourage you to consider GSoC 2024 as well :)
